app/packages/operators/src/runtime/hooks/useInvocationRequestQueue.ts

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useRef, useState, useCallback } from "react";
+import { InvocationRequestQueue, InvocationRequest } from "../operators";
+
+type UseInvocationRequestQueueReturn = {
+  requests: InvocationRequest[];
+  onSuccess: (id: string) => void;
+  onError: (id: string) => void;
+};
+
+/**
+ * useInvocationRequestQueue
+ *
+ * A hook to manage the queue of invocation requests, providing methods to handle success and error events.
+ *
+ * @returns {UseInvocationRequestQueueReturn} - An object containing requests, success, and error handling functions.
+ */
+export default function useInvocationRequestQueue(): UseInvocationRequestQueueReturn {
+  const ref = useRef<InvocationRequestQueue>();
```

Comment:
What is the lifecycle of the queue?

Reply:
I've added [a readme](https://github.com/voxel51/fiftyone/blob/5ac6ec44461b071fcfc81628912c8f3dff081f6c/app/packages/operators/src/runtime/README.md) to answer this question!